pkg/provision/storage/perWorkspaceStorage.go

```diff
@@ -0,0 +1,231 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package storage
+
+import (
+	"errors"
+	"fmt"
+
+	dw "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	"github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"github.com/devfile/devworkspace-operator/pkg/common"
+	"github.com/devfile/devworkspace-operator/pkg/config"
+	"github.com/devfile/devworkspace-operator/pkg/constants"
+	devfileConstants "github.com/devfile/devworkspace-operator/pkg/library/constants"
+	nsconfig "github.com/devfile/devworkspace-operator/pkg/provision/config"
+	"github.com/devfile/devworkspace-operator/pkg/provision/sync"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type perWorkspaceStorageProvisioner struct{}
+
+// TODO: Verify needsStorage is correct. Looks good but not sure.
+func (*perWorkspaceStorageProvisioner) NeedsStorage(workspace *dw.DevWorkspaceTemplateSpec) bool {
+	return needsStorage(workspace)
+}
+
+func syncPerWorkspacePVC(workspace *dw.DevWorkspace, clusterAPI sync.ClusterAPI) (*corev1.PersistentVolumeClaim, error) {
+	namespace := workspace.Namespace
+
+	namespacedConfig, err := nsconfig.ReadNamespacedConfig(namespace, clusterAPI)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read namespace-specific configuration: %w", err)
+	}
+
+	// TODO: Should we adjust the storage size?
+	pvcSize := constants.PVCStorageSize
+	if namespacedConfig != nil && namespacedConfig.CommonPVCSize != "" {
+		pvcSize = namespacedConfig.CommonPVCSize
+	}
+
+	// We have to call clusterAPI.Client.Scheme() because the fakeClusterAPI.Scheme used in testing returns nil
+	pvc, err := getPerWorkspacePVCSpec(workspace, clusterAPI.Client.Scheme(), pvcSize)
+	if err != nil {
+		return nil, err
+	}
+
+	currObject, err := sync.SyncObjectWithCluster(pvc, clusterAPI)
+	switch t := err.(type) {
+	case nil:
+		break
+	case *sync.NotInSyncError:
+		return nil, &NotReadyError{
+			Message: "Updated per-workspace PVC on cluster",
+		}
+	case *sync.UnrecoverableSyncError:
+		return nil, &ProvisioningError{
+			Message: "Failed to sync per-workspace PVC to cluster",
+			Err:     t.Cause,
+		}
+	default:
+		return nil, err
+	}
+
+	currPVC, ok := currObject.(*corev1.PersistentVolumeClaim)
+	if !ok {
+		return nil, errors.New("tried to sync per-workspace PVC to cluster but did not get a PVC back")
+	}
+
+	return currPVC, nil
+
+}
+
+func getPerWorkspacePVCSpec(workspace *dw.DevWorkspace, scheme *runtime.Scheme, size string) (*corev1.PersistentVolumeClaim, error) {
+	// TODO: Just make these part of the function parameters instead?
+	namespace := workspace.Namespace
+	DevWorkspaceId := workspace.Status.DevWorkspaceId
```

Comment:
Generally, vars names aren't capitalized. In Go, capitalized names are exported (except within functions where it doesn't do anything)